dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/dataitems/DataItemQueryTests.java

```diff
@@ -0,0 +1,297 @@
+ * based on the file "setup/metadata.json" => "programIndicators",
+ * "dataElements".
+ * 
+ * The test cases using default pagination will imply "paging=true", which is
+ * the default when "paging" is omitted.
+ *
+ * @author maikel arabori
+ */
+public class DataItemQueryTests extends ApiTest
+{
+    private static final int FOUND = 302;
+
+    private static final int NOT_FOUND = 404;
+
+    private static final int CONFLICT = 409;
+
+    private DataItemActions dataItemActions;
+
+    @BeforeAll
+    public void before()
+    {
+        dataItemActions = new DataItemActions();
+
+        login();
+    }
+
+    @Test
+    public void testGetAllDataItemsUsingDefaultPagination()
+    {
+        // When
+        final ApiResponse response = dataItemActions.get();
+
+        // Then
+        response.validate().statusCode( is( FOUND ) );
+        response.validate().body( "pager", isA( Object.class ) );
+        response.validate().body( "nameableObjects", isA( Object.class ) );
+        response.validate().body( "nameableObjects", is( not( empty() ) ) );
+        assertThat( hasMultipleDataItemTypes( response ), is( true ) );
+    }
+
+    @Test
+    public void testGetAllDataItemsWithoutPagination()
+    {
+        // Given
+        final String noPagination = "?paging=false";
+
+        // When
+        final ApiResponse response = dataItemActions.get( noPagination );
+
+        // Then
+        response.validate().statusCode( is( FOUND ) );
+        response.validate().body( "pager", is( nullValue() ) );
+        response.validate().body( "nameableObjects", isA( Object.class ) );
+        response.validate().body( "nameableObjects", is( not( empty() ) ) );
+        assertThat( hasMultipleDataItemTypes( response ), is( true ) );
+    }
+
+    @Test
+    public void testGetAllDataItemsUsingDefaultPaginationOrderedByCode()
+    {
+        // When
+        final ApiResponse response = dataItemActions.get( "?order=code:asc" );
+
+        // Then
+        response.validate().statusCode( is( FOUND ) );
+        response.validate().body( "pager", isA( Object.class ) );
+        response.validate().body( "nameableObjects", isA( Object.class ) );
+        response.validate().body( "nameableObjects[0].dimensionItemType", isA( String.class ) );
+        response.validate().body( "nameableObjects[0].code", is( "AAAAAAA-1234" ) );
+    }
+
+    @Test
+    public void testFilterByDimensionTypeUsingDefaultPagination()
+    {
+        // Given
+        final String theDimensionType = "PROGRAM_INDICATOR";
+        final String theUrlParams = "?filter=dimensionItemType:in:[%s]";
+
+        // When
+        final ApiResponse response = dataItemActions.get( format( theUrlParams, theDimensionType ) );
+
+        // Then
+        response.validate().statusCode( is( FOUND ) );
+        response.validate().body( "pager", isA( Object.class ) );
+        response.validate().body( "nameableObjects[0].dimensionItemType", is( theDimensionType ) );
```

Comment:
I would validate that every item in nameableObjects array is equal to theDimensionType here. 
For example: `response.validate().body( "nameableObjects.dimensionItemType", everyItem(is( theDimensionType ) ));`

Reply:
Done